File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        map<int,int>mp;
        for(auto u:nums){
            mp[u]++;
        }
        int x = nums.size()/2;
        int ans = 0;
        for(auto u:mp){
            if(u.second % 2 == 1)ans++;
        }
        //cout << ans<<endl;
        return ans <= 0 ;
    }
};

File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.cs
public class Solution {
    public bool DivideArray(int[] nums) {
        int [] Frequency = new int[505];
        foreach(var u in nums) Frequency[u]++;
        foreach(var u in Frequency) {
            if(u % 2 == 1) return false;
        }
        return true;
    }
}